src/query/exceptions.hpp

```diff
@@ -112,16 +112,39 @@ class QueryRuntimeException : public QueryException {
   using QueryException::QueryException;
 };
 
+enum AbortReason : uint8_t {
+  NO_ABORT = 0,
+  TERMINATED = 1,
+  SHUTDOWN = 2,
+  TIMEOUT = 3,
+};
```

Comment:
Could you give a short description what each `AbortReason` is + maybe mention when it happens in real life?